src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -87,21 +96,24 @@ class AppsResource @Inject() (
         .map(_ => throw ConflictingChangeException(s"An app with id [${app.id}] already exists."))
         .getOrElse(app)
 
-      val plan = result(groupManager.updateApp(app.id, createOrThrow, app.version, force))
-
-      val appWithDeployments = AppInfo(
-        app,
-        maybeCounts = Some(TaskCounts.zero),
-        maybeTasks = Some(Seq.empty),
-        maybeDeployments = Some(Seq(Identifiable(plan.id)))
-      )
-
-      maybePostEvent(req, appWithDeployments.app)
-      Response
-        .created(new URI(app.id.toString))
-        .header(RestResource.DeploymentHeader, plan.id)
-        .entity(jsonString(appWithDeployments))
-        .build()
+      val groupUpdateResult = groupManager.updateApp(app.id, createOrThrow, app.version, force).map { plan =>
```

Comment:
This is a little unexpected. Why not the following?

```scala
val plan = await(groupManager.updateApp(app.id, createOrThrow, app.version, force))
```

This would reduce this diff to a minimal.

Reply:
Great point, I'll making this change.

Reply:
You're already in the `async`, so why not using `await` instead of `map`?

Reply:
^ see my comment right above yours :-)